### PR TITLE
Create text input

### PR DESCRIPTION
**Pull request by gaidds**
Created basic text input.

### CODE REVIEW

File: src/components/text-input/Text-input.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        placeholder={type === 'password' ? 'Enter password' : ''}
+        isInvalid={type === 'email' && !!emailError} // Mark input as invalid if there's an email error
+      />
+      {type === 'email' && emailError && (
+        <Alert variant="danger">{emailError}</Alert>
+      )}
+      {type === 'password' && (
+        <Form.Text>
+          <br></br>
+          <Button variant="outline-primary" onClick={togglePasswordVisibility}>
```
Comment (Laaad) [inline src/components/text-input/Text-input.tsx:48]: If it's possible here to change the colour of the button to success, that would be great.

**Review by Laaad — APPROVED** [review]

File: src/components/text-input/types.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export interface TextInputProps {
+  label: string;
+  value: string;
+  onChange: (value: string) => void;
+  type?: 'text' | 'password' | 'email';
```
Comment (Ashraf-Alshashaa) [inline src/components/text-input/types.ts:5]: Do u think that we need to have number type also for fields such as price field?

File: src/components/text-input/Text-input.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      setEmailError('Please enter a valid email address');
+    } else {
+      setEmailError(null);
+    }
+  };
+
+  return (
+    <Form.Group controlId={`formBasic${label}`} className={className}>
+      <Form.Label>{label}</Form.Label>
+      <Form.Control
```
Comment (Ashraf-Alshashaa) [inline src/components/text-input/Text-input.tsx:34]: Do we need to handle required field?

File: src/components/text-input/Text-input.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        placeholder={type === 'password' ? 'Enter password' : ''}
+        isInvalid={type === 'email' && !!emailError} // Mark input as invalid if there's an email error
+      />
+      {type === 'email' && emailError && (
+        <Alert variant="danger">{emailError}</Alert>
+      )}
+      {type === 'password' && (
+        <Form.Text>
+          <br></br>
+          <Button variant="outline-success" onClick={togglePasswordVisibility}>
```
Comment (Ashraf-Alshashaa) [inline src/components/text-input/Text-input.tsx:47]: What do u think about display an eye icon on the text field at the right side instead of the btn?

**Pull request merged**